### PR TITLE
Add download app button for mobile

### PR DESCRIPTION
**Pull request by GoOz**
On mobile view, there's now a button just below the cozy-bar to invite you to download some Cozy apps :)

Please review with care, I'm not 100% sure i took the best path. ;)

⚠️ Needs a new release of Cozy-UI

### CODE REVIEW

**Comment by GoOz** [comment]
[(2) [FIL] Ajouter un lien vers le client desktop et le client mobile dans les apps Drive et Photos](https://trello.com/c/c8iAxpR9/291-(2)-%5Bfil%5D-ajouter-un-lien-vers-le-client-desktop-et-le-client-mobile-dans-les-apps-drive-et-photos)

File: src/components/Main.jsx [hunk 1]
```diff
@@ -2,8 +2,11 @@ import styles from '../styles/main'
 
 import React from 'react'
 
+import ButtonClientMobile from './ButtonClientMobile'
+
 const Main = ({ children }) => (
   <main class={styles['fil-content']}>
+    { __TARGET__ !== 'mobile' && <ButtonClientMobile />}
```
Comment (enguerran) [inline src/components/Main.jsx]: You need to declare the `__TARGET__` variable as a global variable for the lint analysis:

Put on first line
```
/* global __TARGET__ */
```

File: src/components/ButtonClientMobile.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/* eslint-env browser */
```
Comment (enguerran) [inline src/components/ButtonClientMobile.jsx]: Not sure it is needed here.

File: src/components/ButtonClientMobile.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import classNames from 'classnames'
+
+class ButtonClientMobile extends Component {
+  constructor (props) {
+    super(props)
+    const seenAd = localStorage.getItem('app_ad')
+    this.state = {
+      seen: seenAd ? JSON.parse(seenAd) : false
+    }
```
Comment (enguerran) [inline src/components/ButtonClientMobile.jsx]: To clarify the algorithm, you shall write this way:

```js
this.state = {
    seen: JSON.parse(localStorage.getItem('app_ad')) || false
}
```

Then `this.state.seen` takes the value of `JSON.parse(localStorage.getItem('app_ad'))` or `false` if it does not exist.

Another way to write this is with a function:

```js
this.state = {
    seen: getSeenAdState()
}
```

with `getSeenAdState()` as:

```js
function getSeenAdState () {
    const seenAd = localStorage.getItem('app_ad')
    return seenAd ? JSON.parse(seenAd) : false 
}
```

This way the function is responsible to `getSeenAdState` wherever and however the value is stored, retrieved and returned. SRP.

File: src/components/ButtonClientMobile.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  render ({ t }, { seen }) {
+    return (!seen
+      ? <div className={styles['coz-banner-client']}>
+        <a href={t('nav.link-client')} target='_blank' className={styles['coz-btn-clientMobile']} onClick={e => this.read(e)} ><span>{t('nav.btn-client-mobile')}</span></a>
+        <button className={classNames('coz-btn', styles['coz-btn--close'])} onClick={e => this.read()} />
+      </div>
+      : '')
+  }
```
Comment (enguerran) [inline src/components/ButtonClientMobile.jsx]: If you don't want to render anything, your component should return `null`.

Comment (enguerran) [inline src/components/ButtonClientMobile.jsx]: I don't think this has the best readability:

```jsx
render () {
    const { t } = this.props
    const { seen } = this.state
    if (seen) return null

    return (
        <div className={styles['coz-banner-client']}>
            <a href={t('nav.link-client')} target='_blank' className={styles['coz-btn-clientMobile']} onClick={e => this.read(e)} >
                <span>{t('nav.btn-client-mobile')}</span>
            </a>
            <button className={classNames('coz-btn', styles['coz-btn--close'])} onClick={e => this.read()} />
        </div>
    )
}
``` 

or with the preact's `render`'s possibility:

```jsx
render ({ t }, { seen }) {
    if (seen) return null

    return (
        <div className={styles['coz-banner-client']}>
            <a href={t('nav.link-client')} target='_blank' className={styles['coz-btn-clientMobile']} onClick={e => this.read(e)} >
                <span>{t('nav.btn-client-mobile')}</span>
            </a>
            <button className={classNames('coz-btn', styles['coz-btn--close'])} onClick={e => this.read()} />
        </div>
    )
}
```

File: src/components/Main.jsx (hunk 1 above)
Comment (GoOz) [inline src/components/Main.jsx]: Oh yeah thanks. :)

File: src/components/ButtonClientMobile.jsx (hunk 2 above)
Comment (GoOz) [inline src/components/ButtonClientMobile.jsx]: If I don't, the linter squeaks on the use of localStorage

File: src/components/ButtonClientMobile.jsx (hunk 3 above)
Comment (GoOz) [inline src/components/ButtonClientMobile.jsx]: I see. Thanks!

File: src/components/ButtonClient.jsx [hunk 5]
```diff
@@ -5,7 +5,7 @@ import { translate } from '../lib/I18n'
 import React from 'react'
 
 const ButtonClient = ({ t }) => (
-  <a href={t('nav.link-client')} className={styles['coz-btn-client']}><span>{t('nav.btn-client')}</span></a>
+  <a href={t('nav.link-client')} target='_blank' className={styles['coz-btn-client']}><span>{t('nav.btn-client')}</span></a>
```
Comment (kosssi) [inline src/components/ButtonClient.jsx]: 👍

File: src/components/ButtonClientMobile.jsx (hunk 2 above)
Comment (kosssi) [inline src/components/ButtonClientMobile.jsx]: maybe `/* global localStorage */` ?

Comment (GoOz) [inline src/components/ButtonClientMobile.jsx]: Honestly, i don't know. ESLint doc said this but I'm open to suggestions. Do you guys think `/* global localStorage */` is more relevant?

Comment (enguerran) [inline src/components/ButtonClientMobile.jsx]: In fact, you could get rid of the linter's alert by using `window.localStorage`.
I understand the needs, but it is not easy to understand that this comments is for `localStorage`.

**Pull request merged**